arviz/data/io_dict.py

```diff
@@ -223,11 +307,23 @@ def from_dict(
     observed_data : dict
     constant_data : dict
     predictions_constant_data: dict
+    warmup_posterior : dict
+    warmup_posterior_predictive : dict
+    warmup_predictions : dict
+    warmup_log_likelihood : dict
+    warmup_sample_stats : dict
+    save_warmup : bool
+        Save warmup iterations InferenceData object. If not defined, use default
+        defined by the rcParams.
     coords : dict[str, iterable]
         A dictionary containing the values that are used as index. The key
         is the name of the dimension, the values are the index values.
     dims : dict[str, List(str)]
         A mapping from variables to a list of coordinate names for the variable.
+    pred_dims : dict[str, List(str)]
+        A mapping from variables to a list of coordinate names for predictions.
```

Comment:
I know `from_pyro` is missing them, but `pred_coords` should also be present. At least pymc3 via the `from_pymc3_predictions` allows having dimensions with the same name in predictions and "non-predictions" while having different coordinate values. What I'm not so sure is if we should use `pred_dims` and `pred_coords` alone or it would be better to use `pred_dims` and `{**coords, **pred_coords}` so that repeated dims are overriden from coords using pred_coords but coords is still available

Reply:
I think it'll be better to go with the second option here.

Reply:
sounds good